cmd/wallet-monitor/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"math/big"
+	"net/http"
+	"os"
+	"time"
+
+	"contrib.go.opencensus.io/exporter/prometheus"
+	"github.com/urfave/cli/v2"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/types"
+	cliutil "github.com/filecoin-project/lotus/cli/util"
+)
+
+var (
+	log = logging.Logger("main")
+
+	keyAddress  = tag.MustNewKey("address")
+	balMetric   = stats.Int64("actor/balance", "actor blance", "FIL")
+	nonceMetric = stats.Int64("actor/nonce", "nonce", "n")
+	balanceView = &view.View{
+		Name:        "actor/balance",
+		Measure:     balMetric,
+		Aggregation: view.LastValue(),
+		TagKeys:     []tag.Key{keyAddress},
+	}
+	nonceView = &view.View{
+		Name:        "actor/nonce",
+		Measure:     nonceMetric,
+		Aggregation: view.LastValue(),
+		TagKeys:     []tag.Key{keyAddress},
+	}
+)
+
+func main() {
+	app := &cli.App{
+		Name:        "wallet-monitor",
+		Usage:       "monitor wallet addresses",
+		Version:     build.UserVersion(),
+		Description: "Export wallet balances as metrics",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "listen",
+				Value: "0.0.0.0:8888",
+			},
+		},
+		Action: func(cctx *cli.Context) error {
+			api, closer, err := cliutil.GetFullNodeAPI(cctx)
+			if err != nil {
+				return err
+			}
+			defer closer()
+
+			if err := view.Register(balanceView, nonceView); err != nil {
+				return err
+			}
+
+			pe, err := prometheus.NewExporter(prometheus.Options{
+				Namespace: "walletmonitor",
+			})
+			if err != nil {
+				log.Fatalw("failed to create the Prometheus stats exporter", "err", err)
+			}
+
+			go func() {
+				mux := http.NewServeMux()
+				mux.Handle("/metrics", pe)
+				if err := http.ListenAndServe(cctx.String("listen"), mux); err != nil {
+					log.Fatalw("failed to run endpoint", "err", err)
+				}
+			}()
+
+			for range time.Tick(time.Minute) {
```

Comment:
This interval would be good to have as a flag with a environment var backing it.

https://pkg.go.dev/github.com/urfave/cli/v2#DurationFlag

You can then default it to `N * build.BlockDelaySecs`.

Reply:
When using the duration flag, the default value would have to be a duration as well, which BlockDelaySecs is not, so you'd need to convert.